idx/cassandra/cassandra.go

```diff
@@ -236,39 +249,51 @@ func (c *CasIdx) Add(data *schema.MetricData, partition int32) error {
 func (c *CasIdx) rebuildIndex() {
 	log.Info("cassandra-idx Rebuilding Memory Index from metricDefinitions in Cassandra")
 	pre := time.Now()
-	defs := make([]schema.MetricDefinition, 0)
+	var defs []schema.MetricDefinition
 	for _, partition := range cluster.ThisNode.GetPartitions() {
-		iter := c.session.Query("SELECT id, orgid, partition, name, metric, interval, unit, mtype, tags, lastupdate from metric_idx where partition=?", partition).Iter()
-
-		mdef := schema.MetricDefinition{}
-		var id, name, metric, unit, mtype string
-		var orgId int
-		var partition int32
-		var lastupdate int64
-		var interval int
-		var tags []string
-
-		for iter.Scan(&id, &orgId, &partition, &name, &metric, &interval, &unit, &mtype, &tags, &lastupdate) {
-			mdef.Id = id
-			mdef.OrgId = orgId
-			mdef.Partition = partition
-			mdef.Name = name
-			mdef.Metric = metric
-			mdef.Interval = interval
-			mdef.Unit = unit
-			mdef.Mtype = mtype
-			mdef.Tags = tags
-			mdef.LastUpdate = lastupdate
-			defs = append(defs, mdef)
-		}
-		if err := iter.Close(); err != nil {
-			log.Fatal(4, "Could not close iterator: %s", err.Error())
-		}
+		defs = c.LoadPartition(partition, defs)
 	}
 	c.MemoryIdx.Load(defs)
 	log.Info("Rebuilding Memory Index Complete. Took %s to load %d series", time.Since(pre).String(), len(defs))
 }
 
+func (c *CasIdx) Load(defs []schema.MetricDefinition) []schema.MetricDefinition {
+	iter := c.session.Query("SELECT id, orgid, partition, name, metric, interval, unit, mtype, tags, lastupdate from metric_idx").Iter()
+	return c.load(defs, iter)
+}
+
+func (c *CasIdx) LoadPartition(partition int32, defs []schema.MetricDefinition) []schema.MetricDefinition {
+	iter := c.session.Query("SELECT id, orgid, partition, name, metric, interval, unit, mtype, tags, lastupdate from metric_idx where partition=?", partition).Iter()
+	return c.load(defs, iter)
+}
+
+func (c *CasIdx) load(defs []schema.MetricDefinition, iter *gocql.Iter) []schema.MetricDefinition {
+	mdef := schema.MetricDefinition{}
+	var id, name, metric, unit, mtype string
+	var orgId int
+	var partition int32
+	var lastupdate int64
+	var interval int
```

Comment:
if `string` type variables are consolidated on a single line then the `int` type ones could be on one line too